packages/fs-detectors/src/monorepos/get-monorepo-default-settings.ts

```diff
@@ -0,0 +1,130 @@
+import { join } from 'path';
+import { monorepoManagers } from './monorepo-managers';
+import { packageManagers } from '../package-managers/package-managers';
+import { DetectorFilesystem } from '../detectors/filesystem';
+import { detectFramework } from '../detect-framework';
+import JSON5 from 'json5';
+
+export class MissingBuildPipeline extends Error {
+  constructor() {
+    super(
+      'Missing required `build` pipeline in turbo.json or package.json Turbo configuration.'
+    );
+  }
+}
+
+export class MissingBuildTarget extends Error {
+  constructor() {
+    super(
+      'Missing required `build` target in either nx.json, project.json, or package.json Nx configuration.'
+    );
+  }
+}
+
+export async function getMonorepoDefaultSettings(
+  projectName: string,
+  projectPath: string,
+  relativeToRoot: string,
+  detectorFilesystem: DetectorFilesystem
+) {
+  const [monorepoManager, packageManager] = await Promise.all([
+    detectFramework({
+      fs: detectorFilesystem,
+      frameworkList: monorepoManagers,
+    }),
+    detectFramework({
+      fs: detectorFilesystem,
+      frameworkList: packageManagers,
+    }),
+  ]);
+
+  if (monorepoManager === 'turbo') {
+    const [turboJSONBuf, packageJSONBuf] = await Promise.all([
+      detectorFilesystem.readFile('turbo.json').catch(() => null),
+      detectorFilesystem.readFile('package.json').catch(() => null),
+    ]);
+
+    let hasBuildPipeline = false;
+
+    if (turboJSONBuf !== null) {
+      const turboJSON = JSON5.parse(turboJSONBuf.toString('utf-8'));
+
+      if (turboJSON?.pipeline?.build) {
+        hasBuildPipeline = true;
+      }
+    } else if (packageJSONBuf !== null) {
+      const packageJSON = JSON.parse(packageJSONBuf.toString('utf-8'));
+
+      if (packageJSON?.turbo?.pipeline?.build) {
+        hasBuildPipeline = true;
+      }
+    }
+
+    if (!hasBuildPipeline) {
+      throw new MissingBuildPipeline();
+    }
+
+    return {
+      monorepoManager: 'turbo',
+      buildCommand: `cd ${relativeToRoot} && npx turbo run build --filter=${projectName}...`,
+      installCommand: `cd ${relativeToRoot} && ${packageManager} install`,
+      commandForIgnoringBuildStep: `cd ${relativeToRoot} && npx turbo-ignore`,
+    };
```

Comment:
**suggestion (non-blocking):** extract condition bodies into functions.

I think this would be easier to read if each condition delegated to a function for that purpose.

```
switch (monorepoManager) {
  case 'turbo':
    return getTurboDefaultSettings(detectorFilesystem);
  case 'nx':
    return getNxDefaultSettings(detectorFilesystem);
}

return null;
```

Reply:
definitely a good change. I'd like to land this asap so going to punt this recommendation to the next time. I'll add a ticket to track so I improve it later.